packages/client/src/game/Game.ts

```diff
@@ -0,0 +1,362 @@
+    const lastCard = initialPack.pop();
+    if (lastCard) {
+      /* Открываем карту поле раздачи всех карт игрокам */
+      setTimeout(() => {
+        this.table.addUpcard(lastCard);
+      }, ANIMATION_TIME * START_NUM_CARDS_IN_HAND);
+
+      if (lastCard.action) {
+        setTimeout(() => {
+          this.checkCard(lastCard);
+          this.moveLine();
+        }, ANIMATION_TIME * (START_NUM_CARDS_IN_HAND + 1));
+      } else {
+        this.activePlayerId = 0;
+        this.getActivePlayer().highlight();
+      }
+
+      this.activeColor = lastCard.color;
+    }
+    /* Оставшаяся часть карт будет в закрытой колоде */
+    this.table.setClosePack(initialPack);
+  }
+
+  /* Проверяем, может ли в текущий момент игрок выполнить ход */
+  checkPosiibilityOfAction() {
+    return this.activePlayerId === 0;
+  }
+
+  /* Юзер берёт карту со стола */
+  playerTakeCard(clickPos: Record<string, number>) {
+    const playerLayer = this.handEntities['frontHand'];
+    const playerCards = playerLayer.cards;
+    const { x: xClick, y: yClick } = clickPos;
+
+    /* Если клик по слою стола был не на закрытой колоде, ничего не происходит */
+    if (
+      xClick <= 0 &&
+      xClick >= BASE_WIDTH_CARD &&
+      yClick <= 0 &&
+      yClick >= BASE_HEIGHT_CARD
+    ) {
+      console.log('Чтобы взять карту, нужно кликнуть по закрытой колоде');
+      return;
+    }
+
+    /* Если есть карта, которую можно скинуть, то взять из закрытой колоды нельзя */
+    for (let i = playerCards.length - 1; i >= 0; i--) {
+      const card = playerCards[i];
+
+      if (compareCards(card, this.table.getUpcard(), this.activeColor)) {
+        console.log('Нужно сбросить карту');
+
+        return;
+      }
+    }
+
+    /* Если проверки пройдены, юзер берёт карту */
+    this.takeCard(1);
+  }
+
+  /* Юзер сбрасывет карту с руки */
+  playerDiscardCard(clickPos: Record<string, number>) {
+    console.log('Ход игрока');
+
+    const playerLayer = this.handEntities['frontHand'];
+    const playerCards = playerLayer.cards;
+    const { x: xClick, y: yClick } = clickPos;
+
+    for (let i = playerCards.length - 1; i >= 0; i--) {
+      const card = playerCards[i];
+
+      /* Поиск карты, по которой был клик */
+      if (
+        xClick >= card.x! &&
+        xClick <= card.x! + BASE_WIDTH_CARD &&
+        yClick >= card.y! &&
+        yClick <= card.y! + BASE_HEIGHT_CARD
+      ) {
+        /* Проверка, можно ли сбросить эту карту */
+        if (compareCards(card, this.table.getUpcard(), this.activeColor)) {
+          console.log('Перемещение карты');
+
+          this.discardCard(card);
+        } else {
+          console.log('Карта не подходит');
+        }
+        /* Завершаем цикл поиска карты */
+        return;
+      }
+    }
+  }
+
+  botMove() {
+    console.log('Ход бота');
+    const botLayer = this.getActivePlayer();
+    const botCards = botLayer.cards;
+
+    for (let i = 0; i < botCards.length; i++) {
+      const card = botCards[i];

```

Comment:
Попробуй лучше использовать for of
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of
```suggestion
    for (const card of botCards) {
```